### PR TITLE
Followup after #9156

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please test for slc7_amd64_gcc12

File: xrootd.spec [hunk 1]
```diff
@@ -50,8 +50,10 @@ make %makeprocesses VERBOSE=1
 %install
 cd ../build
 make install
+ln -s lib64/cmake/XRootD/XRootDConfig.cmake share/xrootd/cmake/XRootDConfig.cmake
 %{relocatePy3SitePackages}
 
 %post
 %{relocateConfig}bin/xrootd-config
 %{relocateConfig}share/xrootd/cmake/XRootDConfig.cmake
```
Comment (smuzaffar) [inline xrootd.spec]: @iarspider , this line should be deleted as we get errors at install time
```
Can't open /cvmfs/cms-ci.cern.ch/week0/PR_55b5622d/el9_amd64_gcc12/external/xrootd/5.6.4-5d8d28f3136841109f1882398c2ccc64/share/xrootd/cmake/XRootDConfig.cmake: No such file or directory.
```

**Comment by smuzaffar** [comment]
@iarspider , remove the CARGO_HOME changes. I will update cmsBuild to set it

**Comment by iarspider** [comment]
please test

**Comment by smuzaffar** [comment]
https://github.com/cms-sw/pkgtools/commit/d3fad5e7dde4c62c2bf0fb3ce4af27cc739ef168 now sets CARGO_HOME if a pkg has direct/indirect deps on `rust`

File: rust.spec [hunk 2]
```diff
@@ -27,3 +27,9 @@ chmod 0755 %{i}/lib/libLLVM*stable
 
 #Remove doc/man
 rm -rf %{i}/share
+rm -f %{i}/lib/rustlib/install.log
+
+%{relocateConfig}lib/rustlib/manifest-rust-std-x86_64-unknown-linux-gnu
+%{relocateConfig}lib/rustlib/manifest-cargo
+%{relocateConfig}lib/rustlib/manifest-rustc
```
Comment (smuzaffar) [inline rust.spec:35]: @iarspider , these `%{relocateConfig}` lines should be in `%post` section

File: json.spec [hunk 3]
```diff
@@ -19,3 +19,6 @@ make %makeprocesses VERBOSE=1
 %install
 cd ../build
 make install
+
+%{relocateConfig}share/pkgconfig/nlohmann_json.pc
```
Comment (smuzaffar) [inline json.spec:24]: this should be done via `%define PipPostPost `

File: rapidjson.spec [hunk 4]
```diff
@@ -15,10 +15,11 @@ cmake ../%{n}-%{realversion} \
...
     -DRAPIDJSON_BUILD_EXAMPLES=OFF
-  
+
 make %{makeprocesses}
 
 %install
 cd ../build
 make install
 
+%{relocateConfig}lib/cmake/RapidJSON/RapidJSONConfig.cmake
```
Comment (smuzaffar) [inline rapidjson.spec:26]: thsi should be in `%post`

File: xrootd.spec [hunk 5]
```diff
@@ -50,8 +50,9 @@ make %makeprocesses VERBOSE=1
 %install
 cd ../build
 make install
+ln -s lib64/cmake/XRootD/XRootDConfig.cmake share/xrootd/cmake/XRootDConfig.cmake
```
Comment (smuzaffar) [inline xrootd.spec]: why this symlink?

File: json.spec (hunk 3 above)
Comment (smuzaffar) [reply json.spec:24]: pls do it using `%define PipPostPost`

Comment (smuzaffar) [reply json.spec:24]: sorry , yes it should be done in %post

File: xrootd.spec (hunk 5 above)
Comment (iarspider) [reply xrootd.spec]: To keep the file in the old location as well. I can remove that.

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
@smuzaffar I'm a bit confused by this error: I see that this package was installed:
```
18:22:15 Done external+json+3.11.3-b46e2b2561069b3e02d0f04ed9933293
```
so why it is not found later when installing `conifer`?

**Pull request merged**